### DISCUSSION TITLE
How can I use generic keys in `RouteNamedMap`

### DISCUSSION THREAD

**Discussion by Garine519** [posva/unplugin-vue-router#359] [Questions and Help]
I would like `RouteNamedMap` interface to use generic route names in addition to the routes defined in `typed-router.d.ts`
Is there a way I could achieve this?
```
export interface RouteNamedMap2 {
  [key: string]: RouteRecordInfo<string, string, Record<string, string>, Record<string, string>>;
}

type C = RouteNamedMap2 | RouteNamedMap;

export const correctType: C = {
  route: {
    name: 'd',
    path: 'd',
    meta: {},
    paramsRaw: { '': 'test' },
    params: { '': 'test' }
  }
};

export const incorrectType: RouteNamedMap = {
  route: {
    name: 'd',
    path: 'd',
    meta: {},
    paramsRaw: { '': 'test' },
    params: { '': 'test' }
  }
};

```
<img width="941" alt="Screenshot 2024-04-07 at 12 28 02 PM" src="https://github.com/posva/unplugin-vue-router/assets/37899422/6efe35f9-b58b-40e2-b8fa-11969fa35ae3">

### I know we can't do interface unions but would love to get the types AND be able to use generic routes. 

<img width="529" alt="Screenshot 2024-04-07 at 12 29 02 PM" src="https://github.com/posva/unplugin-vue-router/assets/37899422/108253d2-c0fc-47d3-b3a8-e0fa98f4ee5a">

**Comment by posva** [comment]
What are you trying to achieve exactly? You can also add routes at build time with `extendRoute()` and `beforeWriteFiles` in your vite config
